gems/smithy-json/lib/smithy-json/serializer.rb

```diff
@@ -11,10 +11,10 @@ def initialize(options = {})
       end
 
       def serialize(shape, data)
-        return nil if shape == Prelude::Unit
-
         ref = shape.is_a?(ShapeRef) ? shape : ShapeRef.new(shape: shape)
-        ::JSON.dump(shape(ref, data))
+        return nil if ref.shape == Prelude::Unit
```

Comment:
Nit: I don't think we need to explicitly return `nil` here, right?

Reply:
No. I can remove this in the next change.